backend/app/Repositories/EloquentTableSessionRepository.php

```diff
@@ -28,4 +29,32 @@ public function getAvgSessionsTimeByParameter(DatePeriod $datePeriod, string $pa
                 );
             });
     }
+
+    public function groupByParameter(string $parameter, int $website_id, DatePeriod $datePeriod): Collection
+    {
+        $sessions = Session::whereDateBetween($datePeriod)
+        ->forWebsite($website_id)
+        ->with([
+                'system:id,os,browser,resolution_width,resolution_height',
+                'visit.geo_position:id,country,city'
+            ])
+        ->get();
+        $sessions->each(function($item) {
```

Comment:
I am not sure it works as expected. Could you write a test that checks that your endpoint grouping sessions by language?